web-server/v0.4/src/components/PageHeader/index.test.js

```diff
@@ -19,4 +37,72 @@ describe('test getBreadcrumb', () => {
     const urlNameList = urlToList('/search').map(url => getBreadcrumb(routerData, url).name);
     expect(urlNameList).toEqual(['search']);
   });
+
+  it('check getBreadcrumb function implementation', () => {
+    const breadcrumbNameMap = { url: { url: 'urlItem' } };
+    const url = 'url';
+    const breadcrumb = breadcrumbNameMap[url];
+    expect(getBreadcrumb(breadcrumbNameMap, url)).toBe(breadcrumb);
+  });
+});
+
+describe('test rendering of TableFilterSelection page component', () => {
+  it('render with empty props', () => {
+    expect(wrapper).toMatchSnapshot();
+  });
+  it('Check functions implementation', () => {
+    const wrapper2 = shallow(<PageHeader {...mockProps} />);
+    const getBreadcrumbDom = jest.spyOn(wrapper.instance(), 'getBreadcrumbDom');
+    const conversionBreadcrumbList = jest.spyOn(wrapper.instance(), 'conversionBreadcrumbList');
+    const getBreadcrumbProps = jest.spyOn(wrapper.instance(), 'getBreadcrumbProps');
+    const conversionFromLocation = jest.spyOn(wrapper.instance(), 'getBreadcrumbProps');
+
+    wrapper2.setProps({ selectedControllers: ['a', 'b', 'c'] });
+    getBreadcrumbDom();
+    expect(wrapper.state('breadcrumb')).toEqual(conversionBreadcrumbList());
+    expect(conversionBreadcrumbList).toHaveBeenCalled();
+
+    expect(getBreadcrumbProps).toHaveBeenCalled();
+
+    // eslint-disable-next-line no-unused-vars
+    const { routes, params, routerLocation, breadcrumbNameMap } = getBreadcrumbProps();
+    expect(wrapper.find(Breadcrumb).props()).not.toBe({});
+    wrapper
+      .find(Breadcrumb)
+      .props()
+      .itemRender('route', 'params', 'routes', 'paths');
+
+    wrapper.setProps({ params: undefined });
+    expect(conversionFromLocation).toHaveBeenCalled();
+  });
+  it('check functions with null props', () => {
+    const conversionBreadcrumbList = jest.spyOn(wrapper.instance(), 'conversionBreadcrumbList');
+    conversionBreadcrumbList();
+    wrapper.setProps({ routes: undefined, location: undefined });
+    expect(wrapper.instance().conversionBreadcrumbList(null)).toBe(null);
+  });
+  it('check with conversionFromProps', () => {
+    const conversionFromProps = jest.spyOn(wrapper.instance(), 'conversionFromProps');
+    wrapper.setProps({ breadcrumbList: ['item-1', 'item-2'], breadcrumbSeparator: [','] });
+    conversionFromProps();
+  });
+  it('check rendering and change simulation', () => {
+    expect(wrapper.find(Tabs).length).toEqual(1);
+    expect(wrapper.find(TabPane)).toHaveLength(2);
+    wrapper
+      .find(Tabs)
+      .at(0)
+      .simulate('change', 'key1');
+    expect(wrapper.instance().props.onTabChange).toHaveBeenCalled();
+  });
+  it('test the titleHeader', () => {
```

Comment:
For example, naming this test 'should render selected controllers in titleheader' describes the behavior of this test case clearly by letting us know what specific behavior is expected.